src/server/replica.cc

```diff
@@ -1201,29 +1220,30 @@ bool Replica::TransactionData::IsGlobalCmd() const {
 
 // Expired entries within MULTI...EXEC sequence which belong to different database
 // should be executed outside the multi transaciton.
-bool Replica::TransactionReader::ReturnEntryOOO(const journal::ParsedEntry& entry) {
-  return !tx_data_.commands.empty() && entry.opcode == journal::Op::EXPIRED &&
-         tx_data_.dbid != entry.dbid;
+bool Replica::TransactionReader::ReturnEntryOOO(const TransactionData& tx_data,
+                                                const journal::ParsedEntry& entry) {
+  return !tx_data.commands.empty() && entry.opcode == journal::Op::EXPIRED &&
+         tx_data.dbid != entry.dbid;
 }
 
 auto Replica::TransactionReader::NextTxData(JournalReader* reader, Context* cntx)
     -> optional<TransactionData> {
   io::Result<journal::ParsedEntry> res;
+  TransactionData next_tx;
```

Comment:
I fixed the flow in this function, as std::move does not guarantees to empty the data (line 1246) from struct, we need to use swap